src/pages/PasswordReset/ChangePassword.jsx

```diff
@@ -0,0 +1,114 @@
+
+class ChangePassword extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      newpassword: '',
+      confirmpassword: '',
+    };
+    this.onChange = this.onChange.bind(this);
+    this.handleclick = this.handleclick.bind(this);
+  }
+
+  onChange(e) {
+    this.setState({ [e.target.name]: e.target.value });
+  }
+
+  handleclick() {
+    const { newpassword, confirmpassword } = this.state;
+    const { changePassword, match } = this.props;
+
+    changePassword(
+      newpassword,
+      confirmpassword,
+      match.params.token,
+    );
+  }
+
+  render() {
+    const { newpassword, confirmpassword } = this.state;
+    return (
+      <div>
+        <div className="modal-dialog" role="document">
+          <div className="modal-content customise_model">
+            <div className="modal-header">
+              <h3 className="modal-title" id="exampleModalLabel">
+                Authors Haven Password Reset
+              </h3>
+            </div>
+            <div className="modal-body">
+              <div className="password_reset">
+                <div className="form-group">
+                  <label
+                    htmlFor="password"
+                    className="col-form-label form_label"
+                  >
+                    New Password
+                    <input
+                      type="password"
+                      className="form-control form-password"
+                      id="newpassword"
+                      name="newpassword"
+                      value={newpassword}
+                      onChange={e => this.onChange(e)}
+                      placeholder="Enter your password"
+                      required
+                    />
+                    <br />
+                  </label>
+                  <label
+                    htmlFor="password"
+                    className="col-form-label form_label"
+                  >
+                    Confirm Password
+                    <input
+                      type="password"
+                      className="form-control form-password"
+                      id="confirmpassword"
+                      name="confirmpassword"
+                      value={confirmpassword}
+                      onChange={e => this.onChange(e)}
+                      placeholder="Enter your password"
+                      required
+                    />
+                  </label>
+                </div>
+                <div className="modal-footer">
+                  <button
+                    type="button"
+                    onClick={() => this.handleclick()}
+                    className="btn btn-primary"
+                  >
+                    Change password
+                  </button>
+                </div>
+              </div>
+            </div>
+          </div>
+        </div>
+      </div>
+    );
+  }
+}
+
+ChangePassword.propTypes = {
+  changePassword: PropTypes.func,
+  match: PropTypes.func,
```

Comment:
The implementation of match in this component is not as a function but as an object